apps/website/docs/infrastructure/monorepository-setup.md

```diff
@@ -0,0 +1,122 @@
+---
+sidebar_label: Creating a Brand-New Mono-Repository on GitHub Using DX Tools
+---
+
+# Creating a Brand-New Mono-Repository on GitHub Using DX Tools
+
+This guide provides step-by-step instructions for creating and setting up a new
+mono-repository on GitHub using DX tools.
+
+:::info
+
+Currently, only projects with a single environment powered by Azure are
+supported. Additional tooling to support other scenarios is under development.
+
+:::
+
+## Repository Setup
+
+Start by creating the repository on GitHub. Then, follow these steps to complete
+the configuration.
+
+### Repository Access Control
+
+Ensure appropriate access control for your repository:
+
+- [Grant access](https://github.com/orgs/pagopa/repositories?type=source&q=eng-github) to your peers who need it.
+- Always provide `Admin` access to:
+  - Cloud engineers (`engineering-team-cloud-eng`) to ensure future support for
+    DX tooling.
+  - The BOT user associated with your product.
+
+### CODEOWNERS Definition
+
+Define a `CODEOWNERS` file to manage repository ownership. A common setup is:
+
+- Your team has control over the entire repository (`*`).
+- Both your team and cloud engineers have control over the `infra` and `.github`
+  paths.
+
+Example `CODEOWNERS` file:
+
+```md
+# See https://help.github.com/en/articles/about-code-owners#example-of-a-codeowners-file
+
+* <your-team>
+/infra/ <your-team> @engineering-team-cloud-eng
+.github/ <your-team> @engineering-team-cloud-eng
+```
+
+### Creation of Dotfiles
+
+Create the following dotfiles at the root of your repository:
+
+- `.terraform-version`: Specify the Terraform version to use (typically
```

Comment:
suggestion: we can share a curl / wget command to copy these files from the DX repo, e.g.

`curl -O https://raw.githubusercontent.com/utente/repo/main/{file1.txt,file2.txt,file3.txt}`

Reply:
I thought about this: https://github.com/pagopa/dx/pull/391/commits/cba6dc45c5c6516784f5550a8e85682fd208afc7

To me it's easier to maintain rather the docs itself

Reply:
We can't download these files from DX repo, as they have a slightly different config because of different needs

Reply:
The harsh reality is that we don’t yet have a reference to provide. Upon further reflection, I believe it’s best not to share any references at this time. Dot files will be set up once we have a clear method to do so.

Reply:
It's true, but we can consider to use dx-typescript until we have an alternative. It's better than nothing imho.
A couple of days ago I've opened [the PR](https://github.com/pagopa/dx-typescript/pull/26) to update the dot files

Reply:
Removed everything

Reply:
Proposal: Since we want developers to use development containers as the default way to manage their environments (don't we? 😄), we can take advantage of the "templates" feature to distribute these files.

To do this, we need to create a new template called `base`, similar to our `node` template (but with dotfiles included, minus the JS-specific tools).

Once we have this template, the setup process can be simplified as follows:

```bash
npx @devcontainers/cli templates apply -t ghcr.io/pagopa/devcontainer-templates/base:1
```

Reply:
these files are also used in GitHub Actions